### PR TITLE
New simple parser for Q

### PR DESCRIPTION
**Pull request by dionboles-asym**
New parser for Q ```AND  OR  NOT ``` support 
table for console users for counts

### CODE REVIEW

File: cdapython/simple_parser.py [hunk 1]
```diff
@@ -1,238 +1,355 @@
...
+        return self.query
+
+
+class NULL(Token):
+    lbp = 4
+    query = Query()
+
+    def nud(self, context: Parser) -> Query:
+        right_side = context.expression(self.lbp)
+        self.query.node_type = "NOT"
```
Comment (rfricke-asymmetrik) [inline cdapython/simple_parser.py:260]: I think we want this to be unquoted?

Comment (dionboles-asym) [reply cdapython/simple_parser.py:260]: are you talking about the new type value?

File: cdapython/simple_parser.py [hunk 2]
```diff
@@ -244,11 +361,17 @@ def led(self, left: str, context: Parser) -> Query:
...
+lexer.register_token(NotEq, re.compile(r"(\s+<>+\s)"))
+lexer.register_token(And, re.compile(r"(AND|and)"))
+lexer.register_token(Or, re.compile(r"(OR|or)"))
+lexer.register_token(Eq, re.compile(r"(=)"))
+lexer.register_token(IN, re.compile(r"(IN|in)"))
+lexer.register_token(LIKE, re.compile(r"(LIKE|like)"))
+lexer.register_token(NOT, re.compile(r"(NOT|not)"))
+lexer.register_token(IS_NOT, re.compile(r"((IS|is)\s+(NOT|not))"))
+lexer.register_token(NOT_IN, re.compile(r"((NOT|not)\s+(IN|in))"))
+lexer.register_token(NOT_IN, re.compile(r"((NOT|not)\s+(LIKE|like))"))
```
[the remark this answers is not in the document]
Comment (dionboles-asym) [reply cdapython/simple_parser.py:373]: fixed sorry about that thanks for the catch

File: cdapython/simple_parser.py (hunk 1 above)
Comment (rfricke-asymmetrik) [reply cdapython/simple_parser.py:260]: `node_type` is being passed as `NOT` here, I think we can either just remove this class altogether or change the node_type to `unquoted`

**Review by rfricke-asymmetrik — APPROVED** [review]
Nice!

**Pull request merged**